core/include/vecmem/containers/const_device_array.hpp

```diff
@@ -0,0 +1,18 @@
+/** VecMem project, part of the ACTS project (R&D line)
+ *
+ * (c) 2021 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+#pragma once
+
+// Local include(s).
+#include "vecmem/containers/device_array.hpp"
+
+namespace vecmem {
+
+   /// Class mimicking a host-filled constant @c std::array in "device code"
+   template< typename T, std::size_t N >
+   using const_device_array = device_array< const T, N >;
```

Comment:
I'm honestly not sure if it's worth having these aliases in the namespace, it's not too hard (and in my opinion actually clearer) for the user to set the template variable to `const T`, right? There is no `std::const_vector` in the standard library either. :thinking:

Reply:
I'm thinking of the type of templating here that I had to use in the early version of my demonstrator code.

https://github.com/krasznaa/detray_data_model/blob/main/tests/detector.hpp

I just think that it could come in handy to easily template some "high level EDM type" on this type instead of on `vecmem::device_array`. Like:

```c++
template< template< typename, std::size_t > class array_type = vecmem::array >
struct condition_edm_type {
   array_type< float, 50 > m_data;
};
```

For constant input data, this could just be specialised with:

```c++
condition_edm_type< vecmem::const_device_array > object_in_device_code;
```

This may not pan out in the end, but this sort of a thing was why I kept the "constant types" around.